scripts/morty/morty_pde_ode_compare.py

```diff
@@ -0,0 +1,413 @@
+            in the in-core region.
+
+        Returns
+        -------
+        zs : 1D vector
+            Spatial nodes
+        ts : 1D vector
+            Time nodes
+        N_z_t : 2D matrix
+            Each row represents the spatial concentration
+            at a given point in time
+
+        """
+        conc = self.init_conc
+        if not spatial_vary:
+            S_vec, nu_vec, mu_vec = self._step_source()
+        else:
+            S_vec, nu_vec, mu_vec = self._vary_source()
+        J = np.arange(0, self.nodes)
+        Jm1 = np.roll(J,  1)
+
+        N_z_t = np.zeros((len(self.ts)+1, self.nodes))
+        N_z_t[0, :] = conc
+        conc_mult = 1 - mu_vec * self.dt
+        add_source = S_vec * self.dt
+        for ti, t in enumerate(self.ts):
+            conc = (add_source + conc_mult * conc
+                    + self.lmbda * (conc[Jm1] - conc))
+            N_z_t[ti+1, :] = conc
+
+        return self.zs, self.ts, N_z_t
+
+    def ODE(self):
+        """
+        Solves the ODE form of depletion (no neutronic updates)
+
+        Returns
+        -------
+        conc : list of 1D vectors
+            Each item in the list is the spatial concentration vector
+
+        """
+        conc = []
+        for t in ts:
+            current = (self.init_conc[0] * np.exp(-self.mu1 * t) +
+                       self.S1/self.mu1 * (1 - np.exp(-self.mu1 * t)))
+            conc.append(current)
+        return conc
+
+
+if __name__ == '__main__':
+    no_space_ODE = True
+    spatial_source_variation = False
+    gif = False
+    scaled_flux = True
+    tf = 100  # 324_000
+    nodes = 40
+
+    # MSRE data
+    # https://www.tandfonline.com/doi/epdf/10.1080/00295450.2021.1943122 [1]
+    # https://github.com/openmsr/msre/blob/master/core/docs/msrecore.pdf [2]
+    # https://link.springer.com/article/10.1007/s10967-022-08535-3 [3]
+    incore_frac = 0.33
+    excore_frac = 0.67
+    z1 = 200.66  # [2] #272 #[1]
+    z2 = (z1 / incore_frac) * excore_frac
+
+    # 25,233 cm3/s # [1]
+    # 75,708 cm3/s # [2]
+    core_diameter = 140.335  # [2]
+    fuel_frac = 0.225  # [2]
+    xsarea = fuel_frac * (np.pi * (core_diameter/2)**2)
+    nu = 75708 / xsarea
+    nu1 = nu  # 66666.66 [3]
+    nu2 = nu  # 66666.66 [3]
+    loss_core = 6e12 * 2666886.8E-24
+    isotope = 'test'
+    mu = {'test': [2.1065742176025568e-05 + loss_core, 2.1065742176025568e-05]}
+    S = {'test': [24568909090.909092, 0]}
+    initial_guess = [0, 0]
+    nz = 20
+
+    dz = np.diff(np.linspace(0, z1+z2, nz))[0]
+
+    lmbda = 0.9
+    dt = lmbda * dz / nu1
+    Comp = DiffEqComp(mu, S, isotope, nu1, nu2, z1, z2,
+                      nodes, tf, dt, initial_guess, lmbda)
+    print(f'Spatial nodes: {nodes}')
+    ts = np.arange(0, tf+dt, dt)
+    print(f'Temporal nodes: {len(ts)}')
+    print(f'Net data: {nodes * len(ts)}')
+
+    start = time()
+    zs, ts, concs = Comp.fd_PDE(False)
+    if spatial_source_variation:
+        _, _, concs_vary = Comp.fd_PDE(True)
```

Comment:
`_, _` is not good practice, IMO. I know you're not using these variables, but at least indicate what they are with a variable name.